wrappers/swift/SDWEventProperties.swift

```diff
@@ -0,0 +1,285 @@
+        self._properties[name] = value
+    }
+
+    /**
+    Sets an integer property value for an event with PiiTags.
+
+    - Parameters:
+        - name: Name of the property.
+        - value: An integer that contains the property value.
+        - piiKind: The kind of Personal Identifieable Information (PII), one from the ::ODWPiiKind enum values.
+    */
+    public func setProperty(_ name: String, withUInt64Value value: UInt64, withPiiKind piiKind: ODWPiiKind) {
+        self.setProperty(name, withUInt64Value:value)
+        self.setPiiTag(name, withPiiKind:piiKind)
+    }
+
+    /**
+    Sets a Bool property value for an event.
+
+    - Parameters:
+        - name: Name of the property.
+        - value: A Bool that contains the property value.
+    */
+    public func setProperty(_ name: String, withBoolValue value: Bool) {
+        self._properties[name] = value
+    }
+
+    /**
+    Sets a Bool property value along with PiiTags.
+
+    - Parameters:
+        - name: Name of the property.
+        - value: A Bool that contains the property value.
+        - piiKind: The kind of Personal Identifieable Information (PII), one from the ::ODWPiiKind enum values.
+    */
+    public func setProperty(_ name: String, withBoolValue value: Bool, withPiiKind piiKind: ODWPiiKind) {
+        self.setProperty(name, withBoolValue:value)
+        self.setPiiTag(name, withPiiKind:piiKind)
+    }
+
+    /**
+    Sets a UUID property for an event.
+
+    - Parameters:
+        - name: Name of the property.
+        - value: A UUID that contains the property value.
+    */
+    public func setProperty(_ name: String, withUUIDValue value: UUID) {
+        self._properties[name] = value
+    }
+
+    /**
+    Sets a UUID property for an event.
+
+    - Parameters:
+        - name: Name of the property.
+        - value: A UUID that contains the property value.
+        - piiKind: The kind of Personal Identifieable Information (PII), one from the ::ODWPiiKind enum values.
+    */
+    public func setProperty(_ name: String, withUUIDValue value: UUID, withPiiKind piiKind: ODWPiiKind) {
+        self.setProperty(name, withUUIDValue:value)
+        self.setPiiTag(name, withPiiKind:piiKind)
+    }
+
+    /**
+    Sets a date property for an event.
+
+    - Parameters:
+        - name: Name of the property.
+        - value: A Date that contains the property value.
+    */
+    public func setProperty(_ name: String, withDateValue value: Date) {
+        self._properties[name] = value
+    }
+
+    /**
+    Sets a date property for an event with PiiTags.
+
+    - Parameters:
+        - name: Name of the property.
+        - value: A Date that contains the property value.
+        - piiKind: The kind of Personal Identifieable Information (PII), one from the ::ODWPiiKind enum values.
+    */
+    public func setProperty(_ name: String, withDateValue value: Date, withPiiKind piiKind: ODWPiiKind) {
```

Comment:
Thanks for the point. 
Yes, we have to access c++ via objc from swift.

Though I haven't noticed any interoperability issues while building and running the code. However there are minor changes when it comes to using the enum declared in objc from swift, such as Enum prefix name is removed while using. But since they are minor, i thought of keep using the objc enums rather than creating their duplicate in swift.
also, i have create a task to investigate more about the different compiler and version compatibility between swift-objc and update the source files accordingly.